python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_swarm_group_chat.py

```diff
@@ -40,14 +40,22 @@ async def reset(self) -> None:
         self._current_speaker = self._participant_topic_types[0]
 
     async def select_speaker(self, thread: List[AgentMessage]) -> str:
-        """Select a speaker from the participants based on handoff message."""
-        if len(thread) > 0 and isinstance(thread[-1], HandoffMessage):
-            self._current_speaker = thread[-1].target
-            if self._current_speaker not in self._participant_topic_types:
-                raise ValueError("The selected speaker in the handoff message is not a participant.")
-            return self._current_speaker
-        else:
+        """Select a speaker from the participants based on handoff message.
+        Looks for the last handoff message in the thread to determine the next speaker."""
+        if len(thread) == 0:
             return self._current_speaker
+        for message in reversed(thread):
```

Comment:
So any time the select speaker is called it will walk back the thread until it finds the most recent handoff and go there right?

Reply:
Yep that's correct. This also requires the handoff message to address a valid target. So there is an additional work here https://github.com/microsoft/autogen/issues/4199.

Reply:
Yeah this is nice